db/cryptodata/client.go

```diff
@@ -0,0 +1,92 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package cryptodata
+
+import (
+	"crypto/rsa"
+	"github.com/ligato/cn-infra/db/keyval"
+	"errors"
+	"crypto/rand"
+	"io"
+	"hash"
+	"crypto/sha256"
+)
+
+// Client handles encrypting/decrypting and wrapping data
+type Client interface {
+	// EncryptData encrypts input data using provided public key
+	EncryptData(inData []byte, pub *rsa.PublicKey) (data []byte, err error)
+	// DecryptData decrypts input data
+	DecryptData(inData []byte) (data []byte, err error)
+	// Wrap wraps core broker watcher with support for decrypting encrypted keys
+	Wrap(cbw keyval.CoreBrokerWatcher, decrypter ArbitraryDecrypter) keyval.CoreBrokerWatcher
+}
+
+// ClientConfig is result of converting Config.PrivateKeyFile to PrivateKey
+type ClientConfig struct {
+	// Private key is used to decrypt encrypted keys while reading them from store
+	PrivateKeys []*rsa.PrivateKey
+	// Reader used for encrypting/decrypting
+	Reader io.Reader
+	// Hash function used for hashing while encrypting
+	Hash hash.Hash
+}
+
+// ClientWithConfig implements client and client configuration
+type ClientWithConfig struct {
```

Comment:
This might be too verbose. Let's call this `Client` and the interface `ClientAPI` so it better fits with others.